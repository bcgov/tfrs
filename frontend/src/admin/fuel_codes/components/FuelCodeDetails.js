/*
 * Presentational component
 */
import React from 'react';
import PropTypes from 'prop-types';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';

import history from '../../../app/History';
import * as Lang from '../../../constants/langEnUs';
import { FUEL_CODES } from '../../../constants/routes/Admin';

const FuelCodeDetails = props => (
  <div className="page-fuel-code-details">
    <div className="fuel-code-details">
      <div className="row">
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="fuel-code">Low Carbon Fuel Code:
              <div className="value">{props.item.fuelCode}</div>
            </label>
          </div>
        </div>

        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="fuel-code">Company:
              <div className="value">{props.item.company}</div>
            </label>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="carbon-intensity">Carbon Intensity:
              <div className="value">{props.item.carbonIntensity}</div>
            </label>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="application-date">Application Date:
              <div className="value">{props.item.applicationDate}</div>
            </label>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="effective-date">Effective Date:
              <div className="value">{props.item.effectiveDate}</div>
            </label>
          </div>
        </div>
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="expiry-date">Expiry Date:
              <div className="value">{props.item.expiryDate}</div>
            </label>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="fuel">Fuel:
              <div className="value">{props.item.fuel}</div>
            </label>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="feedstock">Feedstock:
              <div className="value">{props.item.feedstock}</div>
            </label>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="feedstock-location">Feedstock Location:
              <div className="value">{props.item.feedstockLocation}</div>
            </label>
          </div>
        </div>
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="feedstock-miscellaneous">Feedstock Miscellaneous:
              <div className="value">{props.item.feedstockMisc}</div>
            </label>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="facility-location">Fuel Production Facility Location (City, Province/State/Country):
              <div className="value">{props.item.facilityLocation}</div>
            </label>
          </div>
        </div>
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="facility-nameplate">Fuel Production Facility Nameplate Capacity (litres/GJ per year):
              <div className="value">{props.item.facilityNameplate}</div>
            </label>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="feedstock-transport-mode">Feedstock Transport Mode
              <div className="value">{props.item.feedstockTransportMode.join(', ')}</div>
            </label>
          </div>
        </div>
        <div className="col-sm-6">
          <div className="form-group">
            <label htmlFor="fuel-transport-mode">Finished Fuel Transport Mode
              <div className="value">{props.item.fuelTransportMode.join(', ')}</div>
            </label>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="col-sm-6">
          <label htmlFor="former-company-name">Former Company Name:
            <div className="value">{props.item.formerCompany}</div>
          </label>
        </div>
      </div>

      <div className="row">
        <div className="col-sm-6">
          <label htmlFor="approval-date">Approval Date:
            <div className="value">{props.item.approvalDate}</div>
          </label>
        </div>
      </div>
    </div>

    <div className="btn-container">
      <button
        className="btn btn-default"
        onClick={() => history.goBack()}
        type="button"
      >
        <FontAwesomeIcon icon="arrow-circle-left" /> {Lang.BTN_APP_CANCEL}
      </button>

      {props.item.status && props.item.status.status === 'Draft' &&
      <button
        className="btn btn-default"
        type="button"
        onClick={() => history.push(FUEL_CODES.EDIT.replace(':id', props.item.id))}
      >
        <FontAwesomeIcon icon="edit" /> {Lang.BTN_EDIT}
      </button>
      }
    </div>
  </div>
);

FuelCodeDetails.defaultProps = {};

FuelCodeDetails.propTypes = {
  item: PropTypes.shape({
    applicationDate: PropTypes.string,
    approvalDate: PropTypes.string,
    carbonIntensity: PropTypes.string,
    company: PropTypes.string,
    effectiveDate: PropTypes.string,
    expiryDate: PropTypes.string,
    facilityLocation: PropTypes.string,
    facilityNameplate: PropTypes.number,
    feedstock: PropTypes.string,
    feedstockLocation: PropTypes.string,
    feedstockMisc: PropTypes.string,
    feedstockTransportMode: PropTypes.arrayOf(PropTypes.string),
    formerCompany: PropTypes.string,
    fuel: PropTypes.string,
    fuelCode: PropTypes.string,
    fuelTransportMode: PropTypes.arrayOf(PropTypes.string),
    id: PropTypes.number,
    status: PropTypes.shape({
      status: PropTypes.string
    })
  }).isRequired
};

export default FuelCodeDetails;
