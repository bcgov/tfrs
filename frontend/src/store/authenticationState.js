import { loadUser } from 'redux-oidc';
import { put, takeLatest, all } from 'redux-saga/effects';
import userManager from './oidc-usermanager';
import { getLoggedInUser } from '../actions/userActions';
import CONFIG from '../config';

const LOGIN_TRIGGERING_ACTIONS = [
  'redux-oidc/USER_EXPIRED'
];

function triggerLoginFlow (store) {
  const { routing } = store.getState();

  if (routing.location &&
    routing.location.pathname !== '/authCallback') {
    return CONFIG.KEYCLOAK.CUSTOM_LOGIN ? userManager.signinSilent() : userManager.signinRedirect();
  }

  return false;
}

function * getBackendUser () {
  yield put(getLoggedInUser());
}

export default function * authenticationStateSaga (store) {
  userManager.clearStaleState();

  const { routing } = store.getState();

  if (!routing.location || routing.location.pathname !== '/authCallback') {
    loadUser(store, userManager);
  }
  yield all([
    takeLatest('redux-oidc/USER_FOUND', getBackendUser),
    takeLatest(
      action => (LOGIN_TRIGGERING_ACTIONS.includes(action.type)),
      triggerLoginFlow,
      store
    )
  ]);
}
