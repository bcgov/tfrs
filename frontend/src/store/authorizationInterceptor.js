import axios from 'axios';
import userManager from './oidc-usermanager';
import CONFIG from '../config';

function configureAxios () {
  if (CONFIG.KEYCLOAK) {
    console.log('adding token to request');

    axios.interceptors.request.use((config) => {
      const loadconfig = async () => {
        const user = await userManager.getUser();
        return {
          ...config,
          headers: {
            Authorization: `Bearer ${user.id_token}`
          }
        };
      };

      return loadconfig();
    });
  }
}

export default configureAxios;
