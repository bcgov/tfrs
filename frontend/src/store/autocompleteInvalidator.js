import { call, put, takeLatest } from 'redux-saga/effects';
import AutocompleteActionTypes from "../constants/actionTypes/Autocomplete";
import FuelCodeActionTypes from '../constants/actionTypes/FuelCodes';

const INVALIDATING_ACTIONS = [
  FuelCodeActionTypes.ADD_FUEL_CODE,
  FuelCodeActionTypes.RECEIVE_FUEL_CODE,
  FuelCodeActionTypes.RECEIVE_FUEL_CODES,
  FuelCodeActionTypes.UPDATE_FUEL_CODE,
  FuelCodeActionTypes.DELETE_FUEL_CODE
];

function * invalidateCache () {
  yield put({ type: AutocompleteActionTypes.INVALIDATE_AUTOCOMPLETE_CACHE });
}

export default function * sessionTimeoutSaga () {
  yield takeLatest(
    action => (INVALIDATING_ACTIONS.includes(action.type)),
    invalidateCache
  );
}
