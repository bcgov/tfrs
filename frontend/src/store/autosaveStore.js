import {takeEvery, all, call} from 'redux-saga/effects';
import {delay} from "redux-saga";
import axios from "axios";
import * as Routes from "../constants/routes";

function* loadAutosaveData(action) {
  const {key, resolve, reject} = action.payload;

  axios
    .get(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}?key=${key}`)
    .then((response) => {
      try {
        const parsed = JSON.parse(response.data.data);
        resolve(parsed);
      } catch (e) {
        reject(e);
      }
    }).catch((error) => {
    reject(error);
  });
}

function* saveAutosaveData(action) {
  const {key, state, resolve, reject} = action.payload;

  const payload = {
    key,
    data: JSON.stringify(state)
  };

  axios
    .post(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}`, payload)
    .then((response) => {
      resolve();
    }).catch((error) => {
    reject(error);
  });
}

function* clearAutosaveData(action) {
  const {resolve, reject} = action.payload;

  axios
    .post(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}/clear`)
    .then((response) => {
      try {
        resolve();
      } catch (e) {
        reject(e);
      }
    }).catch((error) => {
    reject(error);
  });
}

export default function* autosaveSaga() {

  yield all([
    takeEvery('LOAD_AUTOSAVE_DATA', loadAutosaveData),
    takeEvery('SAVE_AUTOSAVE_DATA', saveAutosaveData),
    takeEvery('CLEAR_AUTOSAVE_DATA', clearAutosaveData)
  ]);
}
