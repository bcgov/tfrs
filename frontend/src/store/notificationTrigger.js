import {call, put, takeLatest} from 'redux-saga/effects';
import {delay} from 'redux-saga';
import {getNotifications, getNotificationsCount} from "../actions/notificationActions";
import UserActionTypes from "../constants/actionTypes/Users";
import NotificationActionTypes from "../constants/actionTypes/Notifications";


const TRIGGERING_ACTIONS = [
  'SERVER_INITIATED_NOTIFICATION_RELOAD',
  UserActionTypes.RECEIVE_LOGGED_IN_USER,
  NotificationActionTypes.SUCCESS_NOTIFICATIONS
];

function* fetchNotifications(store) {
  yield call(delay, 1000); //debounce

  if (store.getState().rootReducer.userRequest.isAuthenticated) {
    yield put(getNotificationsCount());
  }

  if (store.getState().rootReducer.notifications.onNotificationsPage) {
    yield put(getNotifications());
  }

}

export default function* notificationsSaga(store) {
  yield takeLatest(
    action => (TRIGGERING_ACTIONS.includes(action.type)),
    fetchNotifications,
    store
  );
}
