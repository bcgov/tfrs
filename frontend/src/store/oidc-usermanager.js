import { createUserManager } from 'redux-oidc';
import CONFIG from '../config';

const settings = {
  authority: CONFIG.KEYCLOAK.AUTHORITY,
  client_id: CONFIG.KEYCLOAK.CLIENT_ID,
  redirect_uri: CONFIG.KEYCLOAK.CALLBACK_URL,
  automaticSilentRenew: true,
  silent_redirect_uri: `${window.location.protocol}//${window.location.hostname}${window.location.port ? `:${window.location.port}` : ''}/token_renew.html`,
  response_type: 'token id_token'
};

const userManager = createUserManager(settings);

export default userManager;
