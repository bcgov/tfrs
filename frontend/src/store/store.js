import { createStore, compose, applyMiddleware, combineReducers } from 'redux';
import { reducer as toastrReducer } from 'react-redux-toastr';
import { routerReducer, routerMiddleware } from 'react-router-redux';
import { createLogger } from 'redux-logger';
import io from 'socket.io-client';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/reducer';
import { getNotifications } from '../actions/notificationActions';
import { SOCKETIO_URL } from '../constants/routes';

import createOidcMiddleware, { loadUser, processSilentRenew, reducer as OIDCReducer } from 'redux-oidc';
import { persistTargetPathReducer } from '../reducers/persistTargetPathReducer';

import userManager from './oidc-usermanager';
import CONFIG from '../config';
import { getLoggedInUser } from '../actions/userActions';

import persistState from 'redux-localstorage';
import createSocketIoMiddleware from 'redux-socket.io';

const middleware = routerMiddleware(history);
const oidcMiddleware = createOidcMiddleware(userManager);

const socket = io(SOCKETIO_URL);
const socketIoMiddleware = createSocketIoMiddleware(socket, 'socketio/');

const enhancer = compose(persistState(['targetPath'], {key: 'tfrs-state'}));

const combinedReducers = (state = {}, action) => {
  const currentRoute = state.routing || {};
  return {
    toastr: toastrReducer(state.toastr, action),
    oidc: OIDCReducer(state.oidc, action),
    routing: routerReducer(state.routing, action),
    targetPath: persistTargetPathReducer(state.targetPath, {...action, currentRoute}),
    rootReducer: rootReducer(state.rootReducer, action),
  };
};

let allMiddleware = [
  thunk,
  socketIoMiddleware,
  oidcMiddleware,
  middleware
];

if (process.env.NODE_ENV !== 'production') {
  allMiddleware.push(createLogger());
}

const store = createStore(
  combinedReducers,
  applyMiddleware(...allMiddleware),
  enhancer
);

let subscriptionProcessing = false;

store.subscribe(() => {
  const state = store.getState();
  if (!subscriptionProcessing) {
    subscriptionProcessing = true;

    if (state.rootReducer.notifications.serverInitiatedReloadRequested === true) {
      store.dispatch(getNotifications());
    }

    if (CONFIG.KEYCLOAK.ENABLED) {
      if (state.oidc.user &&
        !state.rootReducer.userRequest.isFetching &&
        !state.rootReducer.userRequest.isAuthenticated) {
        store.dispatch(getLoggedInUser());
      }
    }

    subscriptionProcessing = false;
  }
});

if (CONFIG.KEYCLOAK.ENABLED) {
  loadUser(store, userManager);
  processSilentRenew();
}

export default store;
