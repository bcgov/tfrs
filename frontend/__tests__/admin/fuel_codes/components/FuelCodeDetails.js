import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import FuelCodeDetails from '../../../../src/admin/fuel_codes/components/FuelCodeDetails';
import store from '../../../../src/store/store';

test('FuelCodeDetails should display', () => {
  const item = {
    feedstockTransportMode: [],
    fuelTransportMode: []
  };

  const component = renderer.create(
    <Provider store={store}>
      <FuelCodeDetails item={item} />
    </Provider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
