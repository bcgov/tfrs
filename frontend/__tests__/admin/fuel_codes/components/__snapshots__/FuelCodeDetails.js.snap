// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FuelCodeDetails should display 1`] = `
<div
  className="page-fuel-code-details"
>
  <div
    className="fuel-code-details"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="fuel-code"
          >
            Low Carbon Fuel Code:
            <div
              className="value"
            />
          </label>
        </div>
      </div>
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="fuel-code"
          >
            Company:
            <div
              className="value"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="carbon-intensity"
          >
            Carbon Intensity:
            <div
              className="value"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="application-date"
          >
            Application Date:
            <div
              className="value"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="effective-date"
          >
            Effective Date:
            <div
              className="value"
            />
          </label>
        </div>
      </div>
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="expiry-date"
          >
            Expiry Date:
            <div
              className="value"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="fuel"
          >
            Fuel:
            <div
              className="value"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="feedstock"
          >
            Feedstock:
            <div
              className="value"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="feedstock-location"
          >
            Feedstock Location:
            <div
              className="value"
            />
          </label>
        </div>
      </div>
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="feedstock-miscellaneous"
          >
            Feedstock Miscellaneous:
            <div
              className="value"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="facility-location"
          >
            Fuel Production Facility Location (City, Province/State/Country):
            <div
              className="value"
            />
          </label>
        </div>
      </div>
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="facility-nameplate"
          >
            Fuel Production Facility Nameplate Capacity (litres/GJ per year):
            <div
              className="value"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="feedstock-transport-mode"
          >
            Feedstock Transport Mode
            <div
              className="value"
            >
              
            </div>
          </label>
        </div>
      </div>
      <div
        className="col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="fuel-transport-mode"
          >
            Finished Fuel Transport Mode
            <div
              className="value"
            >
              
            </div>
          </label>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <label
          htmlFor="former-company-name"
        >
          Former Company Name:
          <div
            className="value"
          />
        </label>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <label
          htmlFor="approval-date"
        >
          Approval Date:
          <div
            className="value"
          />
        </label>
      </div>
    </div>
  </div>
  <div
    className="btn-container"
  >
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="button"
    >
       
      Back
    </button>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="button"
    >
       
      Edit
    </button>
  </div>
</div>
`;
